magpy/gui/magpy_gui.py

```diff
@@ -2872,6 +2891,48 @@ def onChangeSymbol(self, event):
             self.OnPlot(self.plotstream,self.shownkeylist)
         self.changeStatusbar("Ready")
 
+    def OnFlagClick(self, event):
+        """Mouse event for flagging with double click."""
+        if not event.inaxes or not event.dblclick:
+            return
+        sensid = self.plotstream.header.get('SensorID','')
+        dataid = self.plotstream.header.get('DataID','')
+        if sensid == '' and not dataid == '':
+            sensid = dataid[:-5]
+        if sensid == '':
+            dlg = wx.MessageDialog(self, "No Sensor ID available!\n"
+                            "You need to define a unique Sensor ID\nfor the data set in order to use flagging.\nPlease go the tab Meta for this purpose.\n","Undefined Sensor ID", wx.OK|wx.ICON_INFORMATION)
+            dlg.ShowModal()
+            dlg.Destroy()
+        else:
+            flaglist = []
+            xdata = self.plot_p.t
+            xtol = ((max(xdata) - min(xdata))/float(len(xdata)))/2
+            pickX = event.xdata
+            for x in xdata:
+                if  pickX-xtol < x < pickX+xtol :
+                    starttime = num2date(x - xtol)
+                    endtime = num2date(x + xtol)
+                    dlg = StreamFlagSelectionDialog(None, title='Stream: Flag Selection', shownkeylist=self.shownkeylist, keylist=self.keylist)
+                    if dlg.ShowModal() == wx.ID_OK:
+                        keys2flag = dlg.AffectedKeysTextCtrl.GetValue()
+                        keys2flag = keys2flag.split(',')
+                        keys2flag = [el for el in keys2flag if el in KEYLIST]
+                        flagid = dlg.FlagIDComboBox.GetValue()
+                        flagid = int(flagid[0])
+                        comment = dlg.CommentTextCtrl.GetValue()
+                        if comment == '' and flagid != 0:
+                            comment = 'Point flagged with unspecified reason'
+                        flaglist = self.plotstream.flag_range(keys=self.shownkeylist,flagnum=flagid,text=comment,keystoflag=keys2flag,starttime=starttime,endtime=endtime)
+                        self.menu_p.rep_page.logMsg('- flagged time range: added {} flags'.format(len(flaglist)))
+            if len(flaglist) > 0:
```

Comment:
The `if len(flag list) > 0` was not outdented, so that it would not be evaluated when there isn't a sensor id.